### PR TITLE
Problem: concurrent write base fee in fee history (backport: #364)

### PR DESCRIPTION
**Pull request by mmsqe**
for more [info](https://github.com/crypto-org-chain/ethermint/pull/364)

Closes: #XXX

## Description

______

For contributor use:

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added  relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer

______

For admin use:

- [ ] Added appropriate labels to PR (ex. `WIP`, `R4R`, `docs`, etc)
- [ ] Reviewers assigned
- [ ] Squashed all commits, uses message "Merge pull request #XYZ: [title]" ([coding standards](https://github.com/tendermint/coding/blob/master/README.md#merging-a-pr))

### CODE REVIEW

File: rpc/backend/backend.go [hunk 1]
```diff
@@ -189,4 +199,6 @@ func NewBackend(
 		allowUnprotectedTxs: allowUnprotectedTxs,
 		indexer:             indexer,
 	}
+	b.processBlocker = b.processBlock
```
Comment (yihuang) [inline rpc/backend/backend.go:202]: is this useful?

Comment (mmsqe) [reply rpc/backend/backend.go:202]: need for test

Comment (yihuang) [reply rpc/backend/backend.go:202]: I mean the assignment, `b.processBlock` is always `nil` here

Comment (mmsqe) [reply rpc/backend/backend.go:202]: why? [b.processBlock](https://github.com/mmsqe/ethermint/blob/c0312a4afe2959c0a01d22f2848816e0aeb504a1/rpc/backend/utils.go#L162) is a function

Comment (yihuang) [reply rpc/backend/backend.go:202]:
```
b := &Backend{
```

`b` is just constructed, where the `processBlock` field is not set, right?

Comment (mmsqe) [reply rpc/backend/backend.go:202]: but processBlocker is property, b.processBlock =/= b.processBlocker

Comment (yihuang) [reply rpc/backend/backend.go:202]: oh, `processBlock` is a method.

**Review by yihuang — APPROVED** [review]

**Pull request merged**